app/experimenter/experiments/forms.py

```diff
@@ -243,38 +243,44 @@ class Meta:
         ]
 
 
-class ExperimentVariantsFormSet(BaseInlineFormSet):
+class BaseExperimentVariantsFormSet(BaseInlineFormSet):
+    """This class exists to *share* formset checks on the list of form
+    instances (self.forms) that is common for all types of experiment
+    variants."""
 
-    def clean(self):
+    def clean_forms(self):
         alive_forms = [
-            form for form in self.forms if not form.cleaned_data["DELETE"]
```

Comment:
Note here, in the old code `form.is_valid()` was never called so it'd be illegal to access `form.cleaned_data`. I rearranged things so it becomes moot but still worth noticing.